### PR TITLE
feat(TPG >= 4.46)!: adds support to autoclass

### PR DESCRIPTION
**Pull request by fernandoiury**
This PR provides the autoclass feature for both  multiple_buckets and simple_bucket.

Closes https://github.com/terraform-google-modules/terraform-google-cloud-storage/issues/220

### CODE REVIEW

**Comment by fernandoiury** [comment]
@apeabody I just fixed lint. It's passing locally now. TIA

**Comment by fernandoiury** [comment]
@apeabody sorry I don't know the whole process. Do I have to do anything else or just wait for the reviewers for now? TIA

**Comment by apeabody** [comment]
> @apeabody sorry I don't know the whole process. Do I have to do anything else or just wait for the reviewers for now? TIA

Hi @fernandoiury - Yes, typically wait for the reviewers, but I'll take a quick look at it now.

File: modules/simple_bucket/variables.tf [hunk 1]
```diff
@@ -54,6 +54,12 @@ variable "versioning" {
   default     = true
 }
 
+variable "autoclass" {
+  description = "While set to true, autoclass is enabled for this bucket."
+  type        = bool
+  default     = true
```
Comment (apeabody) [inline modules/simple_bucket/variables.tf:60]: Unless autoclass is currently enabled by default, we should probably default this to `false` to maintain existing behavior (avoid breaking change).

```suggestion
  default     = false
```

File: variables.tf [hunk 2]
```diff
@@ -60,6 +60,12 @@ variable "versioning" {
   default     = {}
 }
 
+variable "autoclass" {
+  description = "Optional map of lowercase unprefixed name => boolean, defaults to false."
```
Comment (apeabody) [inline variables.tf:64]:
```suggestion
  description = "Optional map of lowercase unprefixed bucket name => boolean, defaults to false."
```

**Review by apeabody — CHANGES_REQUESTED** [review]
Thanks for the contribution @fernandoiury!  A few notes:

Please bump the minimum TPG version to the first which includes `autoclass`:
* https://github.com/terraform-google-modules/terraform-google-cloud-storage/blob/master/versions.tf#L23
* https://github.com/terraform-google-modules/terraform-google-cloud-storage/blob/master/modules/simple_bucket/versions.tf#L23

We should also exercise the new option in a test covered by the CI tests.  Basically just enable it for one of these examples:
* https://github.com/terraform-google-modules/terraform-google-cloud-storage/tree/master/examples

**Comment by fernandoiury** [comment]
- Versions bumped
- Added autoclass = true to one of the examples
- fixed docs / updated metadata

**Review by apeabody — APPROVED** [review]
Thanks for the contribution @fernandoiury - LGTM

As this requires a bump to minimum TPG (breaking change), I'll first release `4.0.1`.

**Pull request merged**